### PR TITLE
Formatting chat

### PR DESCRIPTION
**Pull request by CJBuzz**
Implemented formatting of chatbot output

### CODE REVIEW

File: backend/fast_api/api.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ async def chatbot(request: Request):
         session_id = data.get('sessionID')
         top_schemes_text = ""
 
-        df = pd.DataFrame(top_schemes[session_id])
+        df = pd.DataFrame(top_schemes[str(session_id)])
```
Comment (rebeccacxy) [inline backend/fast_api/api.py:186]: not necessary

File: frontend/static/js/events.js [hunk 2]
```diff
@@ -1,83 +1,126 @@
...
+  var segment = $("#filler");
+  segment.html("");
+  $("#filler")
+    .hide()
+    .append(
+      '<div class="text-center"><div class="spinner-border text-primary" role="status"><span class="sr-only">Loading...</span></div></div>'
+    )
+    .fadeIn(1000);
+  // var url = "https://schemes.sg/schemespredict";
+  var url = "http://127.0.0.1:8000/schemespredict";
```
Comment (rebeccacxy) [inline frontend/static/js/events.js:17]: call the frontend api
```suggestion
  var url = "/schemespredict";
```

File: frontend/static/js/passfromprevious.js [hunk 3]
```diff
@@ -1,87 +1,126 @@
...
+    .hide()
+    .append(
+      '<div class="text-center"><div class="spinner-border text-primary" role="status"><span class="sr-only">Loading...</span></div></div>'
+    )
+    .fadeIn(1000);
 
-    // var url = "https://schemes.sg/schemespredict";
-    var url = '/schemespredict'
+  // var url = "https://schemes.sg/schemespredict";
+  var url = "http://127.0.0.1:8000/schemespredict";
```
Comment (rebeccacxy) [inline frontend/static/js/passfromprevious.js:23]: call the frontend api
```suggestion
  var url = "/schemespredict";
```

File: frontend/static/js/chatbot.js [hunk 4]
```diff
@@ -1,32 +1,34 @@
...
-            'Content-Type': 'application/json',
-        },
-        body: JSON.stringify({ data: userInput })
-    });
+  const response = await fetch(url, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ message: userInput }),
```
Comment (rebeccacxy) [inline frontend/static/js/chatbot.js:14]: key should be data
```suggestion
    body: JSON.stringify({ data: userInput }),
```

File: frontend/static/js/chatbot.js [hunk 5]
```diff
@@ -1,32 +1,34 @@
+var url = "http://127.0.0.1:8000/chatbot";
```
[the remark this answers is not in the document]
Comment (rebeccacxy) [reply frontend/static/js/chatbot.js:1]: actually the backend should be running on 0.0.0.0:8000 now

**Comment by CJBuzz** [comment]
I updated chatbot branch based on changes, will make another PR

**Pull request closed without merging**